pkg/services/govmomi/find/object_test.go

```diff
@@ -0,0 +1,191 @@
+package find_test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/vmware/govmomi/simulator"
+	"k8s.io/utils/pointer"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/find"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
+	"sigs.k8s.io/cluster-api-provider-vsphere/test/helpers"
+)
+
+const TestHostGroup = "host-group-alpha"
+
+type resourceCount struct {
+	datacenters                 int
+	clustersInDefaultDatacenter int
+	hostsInTestGroup            int
+}
+
+func TestObjectFunc(t *testing.T) {
+	g := NewWithT(t)
+
+	ctx := context.Background()
+
+	sim, authSession, resources, err := setupSimulatorAndSession(simulator.VPX())
+	g.Expect(err).ToNot(HaveOccurred(), "a vcsim instance and authSession should be established")
+	t.Cleanup(sim.Destroy)
+
+	defaultDatacenter, err := authSession.Finder.DefaultDatacenter(ctx)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	topology := infrav1.Topology{
+		Datacenter:     defaultDatacenter.Name(),
+		ComputeCluster: pointer.String(defaultDatacenter.Name() + "_C0"),
+		Hosts:          &infrav1.FailureDomainHosts{HostGroupName: TestHostGroup},
+	}
+
+	testCases := []struct {
+		name          string
+		failureDomain infrav1.FailureDomainType
+		topology      infrav1.Topology
+		testFunc      func(*WithT, find.ManagedRefFinder)
+	}{
+		{
+			name:          "get and test a finder for compute clusters in the default datacenter",
+			failureDomain: infrav1.ComputeClusterFailureDomain,
+			topology:      topology,
+			testFunc: func(g *WithT, refFinder find.ManagedRefFinder) {
+				computeClusters, err := refFinder(ctx)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(len(computeClusters)).To(Equal(resources.clustersInDefaultDatacenter))
```

Comment:
nit:
```suggestion
				g.Expect(computeClusters).To(HaveLen(resources.clustersInDefaultDatacenter))
```

Reply:
Thanks for reviewing! Resolved with the latest commit.